### ISSUE TITLE
[CLI]: problem with removing the callback in `fastai`

### ISSUE THREAD

**Issue by deven-gqc** [wandb/wandb#4798]
## Environment: In Google Colab
I have a training routine in `fastai` with a `wandb` callback to log the training. Once the training is finished, I remove the callback and then export the model.

Then in a separate notebook, I load the model and use it for inference, and everything works as expected

## Environment: Compute Cluster (with SLURM)

When I use that same training routine in SLURM, remove the callback and then export the model, then during inference, the model starts a log in the `wandb` project. Which is surprising as this behavior is not expected.

```python
dls = ImageDataLoaders(...)
learn = vision_learner(dls, resnet50, cbs=[WandbCallback()])
learn.fine_tune(10)
learn.remove_cb(WandbCallback)
learn.export('foo.pkl')
wandb.finish()
```

```py
Currently running fastai_multi_label_predict_DNV.py with test_0 image_csvs
wandb: Currently logged in as: sengv1. Use `wandb login --relogin` to force relogin
wandb: Appending key for api.wandb.ai to your netrc file: /home/vannary/.netrc
/project/6062137/vannary/CCTV/fastai/lib/python3.7/site-packages/sklearn/metrics/_classification.py:1570: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no true nor predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, "true nor predicted", "F-score is", len(true_sum))
/project/6062137/vannary/CCTV/fastai/lib/python3.7/site-packages/sklearn/metrics/_classification.py:1570: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no true nor predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, "true nor predicted", "F-score is", len(true_sum))
wandb: ERROR You must call wandb.init() before wandb.log()
Traceback (most recent call last):
  File "./script_runner.py", line 75, in <module>
    main()
  File "./script_runner.py", line 65, in main
    script_module.main()
  File "/localscratch/vannary.56580374.0/sewer_ml/script/fastai_multi_label_predict_DNV.py", line 118, in main
    wandb.log({'F1 score': f1_Multi})
  File "/project/6062137/vannary/CCTV/fastai/lib/python3.7/site-packages/wandb/sdk/lib/preinit.py", line 36, in preinit_wrapper
    raise wandb.Error(f"You must call wandb.init() before {name}()")
wandb.errors.Error: You must call wandb.init() before wandb.log()
```

### Environment

WandB version: 0.13.3
OS: Ubuntu
Python version: 3.7.9
Versions of relevant libraries: `fastai`: 2.7.9

### Additional Context

cc @sengv1

**Comment by ArtsiomWB** [comment]
Hi @deven-gqc! Could you send your debug logs in here as well?

**Comment by deven-gqc** [comment]
@ArtsiomWB I'm not sure what other debug logs do you need? Could you please elaborate?

**Comment by morganmcg1** [comment]
Hey @deven-gqc , I think best practice for fastai logging callbacks is to pass the callback to the `.fit` or `.fine_tune` method as opposed to the learner itself. It is strange that `remove_callback` isn't working though.

**Comment by deven-gqc** [comment]
@morganmcg1 Yes, the issue is even more strange it behaves differently in Colab vs a SLURM env.

**Comment by ArtsiomWB** [comment]
@deven-gqc could you try to add wandb.finish before calling remove callback?

The debug logs should be located in the wandb folder in your computer's working directory (or I guess in Colab directory in this case). That folder has folders formatted as run-DATETIME-ID - each of which is associated with an individual run. Could you retrieve the `debug.log` and `debug-internal.log` files from one of these folders?

**Comment by sengv1** [comment]
I found the problem. I tried logging my F1 and F2 scores in my prediction script. However, I did not call `wandb.init()` in my prediction script. I did not edit my prediction script when we fixed the problem with `remove_callback()`  in our training script. 

Before we fixed the problem with `remove_callback()`, it would create a new wandb every time we loaded the model in my prediction script. So, I relied on that to log my F1 and F2 scores. 

The problem is fixed once I called `wandb.init()` from my prediction script.  Thank you, everyone.
